subworkflows/local/alignment.nf

```diff
@@ -86,3 +90,32 @@ workflow ALIGNMENT {
         salmon_info     = SALMON_QUANT.out.json_info
         versions        = ch_versions
 }
+
+
+// Custom functions
+
+/**
+* Branch the read channel into differnt channels,
+* depending on whether the sample has multiple fastq files or not.
+* The resulting channels gets the original sample id in meta.
+*
+* @param ch_reads Channel containing meta and fastq reads
+* @return Channel containing meta with original id and branched on number of fastq files
+*/
+def branchFastqToSingleAndMulti(ch_reads) {
+
+    return ch_reads
+        .map {
+            meta, fastq ->
+                original_id = meta.id.split('_T')[0..-2].join('_')
+                [ meta + [id: original_id], fastq ]
+        }
+        .groupTuple()
+        .branch {
+            meta, fastq ->
+                single_fq: fastq.size() == 1
+                    return [ meta, fastq.flatten() ]
+                multiple_fq: fastq.size() > 1
+                    return [ meta, fastq.flatten() ]
+        }
+}
```

Comment:
It´s true, it would concatenate all the time as it is written. 
They do like this in [RNAseq](https://github.com/nf-core/rnaseq/blob/3bec2331cac2b5ff88a1dc71a21fab6529b57a0f/workflows/rnaseq.nf#L202). It is pretty much the same as you have done, but what if sample id does not end in _T? Could we instead ask for input fastq from R1 to be separated by " "?

Reply:
We could think of changing the T to something else. Right now it is added to the sample name in this function.
https://github.com/genomic-medicine-sweden/tomte/blob/ac85767a74fd8775c04b60764974adc4691b47d6/bin/check_samplesheet.py#L147
So the sample name will always have this suffix. When we switch to nf-validation we can change this :)